includes/class-amp-theme-support.php

```diff
@@ -2033,7 +2033,7 @@ public static function is_output_buffering() {
 	 */
 	public static function finish_output_buffering( $response ) {
 		self::$is_output_buffering = false;
-		return self::prepare_response( $response );
+		return apply_filters( 'amp_document_output', self::prepare_response( $response ) );
```

Comment:
This is a very powerful filter that has the possibility of overriding AMP validity. Any such mutations to the document should rather be done through sanitizers, which the AMP plugin will then guarantee to be valid AMP since the tag-and-attribute sanitizer runs last. So I think bb371b1c0115009d82289c49a72dd6009bababba should be reverted.

Reply:
@westonruter we found this feature to be necessary in-order to integrate non AMP compatible code into our documents. This was a conscious decision on our part & we were well aware of the implications of such a change in terms of AMP validation. As a publisher this level of flexibility is really important as unfortunately AMP compatibility is not a priority for some third-party services.

As an aside I think AMP validation in general needs to be overhauled to provide a broader degree of opt-in/opt-out flexibility but I see that as a larger discussion outside of the scope of this specific PR & better suited to [ampproject/amphtml](https://github.com/ampproject/amphtml).

Reply:
@indieisaconcept What specific changes did you need to make? There is now a way to do this officially via AMP dev mode, depending on what your use case.

Reply:
I've reverted this for now in f59a9c9.